exonum/src/runtime/dispatcher/schema.rs

```diff
@@ -119,6 +119,14 @@ impl<T: Access> Schema<T> {
     }
 }
 
+// `ReadonlyFork` specialization to ensure that we won't leak mutable schema access.
+impl Schema<ReadonlyFork<'_>> {
+    /// Readonly set of launched service instances.
+    pub fn running_instances(&self) -> MapIndex<ReadonlyFork<'_>, String, InstanceSpec> {
```

Comment:
Hm, I thought about literally exposing an iterator:

```rust
pub fn services<'s>(&'s self) -> impl Iterator<Item = InstanceSpec> + 's {
    // ...
}
```

The problem with this approach is that it's seemingly impossible given MerkleDB limitations (it is impossible to have an iterator without a corresponding index, although logically they are independent - the lifetime of the iterator is tied to the lifetime of a snapshot / fork).

An alternative is to use

```rust
impl<T> Schema<T>
where
    T: Access + AsReadonly<Readonly = T>,
{
    // ...
}
```

This seemingly works, although you would need to add `impl AsReadonly for ReadonlyFork` (the implementation is trivial).

Reply:
>Hm, I thought about literally exposing an iterator

I thought about it too, but this approach is worse if some service will want to lookup concrete service: it will have to enjoy `O(N)` approach instead of `O(1)`.

>This seemingly works, although you would need to add impl AsReadonly for ReadonlyFork (the implementation is trivial).

Like it, will do.

Reply:
Done.